common-service/src/main/java/org/sunbird/controllers/BaseController.java

```diff
@@ -240,34 +408,99 @@ public static Response createSuccessResponse(String path, Response response, Str
   }
 
   /**
-   * Determine version of API request based on URI.
+   * Common exception response handler method.
    *
-   * @param path Request URI path
-   * @return Version of API request
+   * @param e Exception represents the exception occurred while processing the request.
+   * @param request play.mvc.Http.Request
+   * @return Result
    */
-  public static String getApiVersion(String path) {
-    if (StringUtils.isBlank(path)) {
-      ProjectLogger.log("BaseController:getApiVersion: Path is blank.", LoggerEnum.INFO);
-      return "v1";
+  public Result createCommonExceptionResponse(
+      Exception e, Request request, boolean generateTelemetry) {
+
+    ProjectCommonException exception = null;
+    if (e instanceof ProjectCommonException) {
+      exception = (ProjectCommonException) e;
+    } else {
+      exception =
+          new ProjectCommonException(
+              ResponseCode.internalError.getErrorCode(),
+              ResponseCode.internalError.getErrorMessage(),
+              ResponseCode.SERVER_ERROR.getResponseCode());
     }
-    return path.split("[/]")[1];
+
+    Request req = request;
+    if (req == null) {
+      req = request();
+    }
+
+    if (generateTelemetry) {
+      generateTelemetryForExceptionResponse(exception, req);
+    }
+    return createCommonExceptionResult(req.path(), exception, req.method());
   }
 
-  /**
-   * Setter method for actor reference or selection object.
-   *
-   * @param actorRefObj Actor reference or selection object
-   */
-  public static void setActorRef(Object actorRefObj) {
-    actorRef = actorRefObj;
+  private String generateStackTrace(StackTraceElement[] elements) {
+    StringBuilder builder = new StringBuilder("");
+    for (StackTraceElement element : elements) {
+
+      builder.append(element.toString());
+      builder.append("\n");
+    }
+    return builder.toString();
+  }
+
+  private void generateTelemetryForExceptionResponse(
+      ProjectCommonException exception, Request request) {
+
+    ProjectLogger.log(exception.getMessage(), exception, generateTelemetryInfoForError());
```

Comment:
Add a comment.

Generate a telemetry event of type ERROR using project logger for exception